Assignment 2

AntonioGomez12 opened the pull request:
link: https://orange-flower-0da9ed11e.1.azurestaticapps.net
Extra credit: added sound to keyboard buttons, added help bar to clarify color and how to play.
partner: Antonio Gomez

A-Tanner commented:
I will review this

A-Tanner commented on wordle-web/scripts/wordOptions.ts:
```diff
@@ -0,0 +1,37 @@
+       return this.words
+     }
+
+     let expression: string = '^[A-Z]{0,5}'
+     for (let i = 0; i < 5; i++) {
+       if (word.charAt(i) === '?') {
+         expression += '[A-Z]'
+       } else {
+         expression += word.charAt(i)
+       }
+     }
```
Expression should be empty at the beginning of this loop.

I was able to get the hint button to display the proper strings by replacing the above code with the following...

```
let expression: string = ''
     let expression: string = ''
     for (let i = 0; i < 5; i++) {
       if (word.charAt(i) === '?') {
         expression += '.'
       } else {
         expression += word.charAt(i)
       }
     }
     }
```

However, some tests were still not passing...

A-Tanner commented on wordle-web/test/wordOptionTest.test.ts:
```diff
@@ -0,0 +1,66 @@
+     expect(allWords[1]).toBeFalsy()
+   })
+
+   test('word A????', () => {
+     words = new wordOptions()
+     word = 'A????'
+     allWords = words.validWords(word)
+     expect(allWords.length).toBeGreaterThan(0)
+     expect(allWords.length).toBe(29)
+     expect(words.count).toBe(29)
+     expect(allWords.includes('ANGEL')).toBeTruthy()
+     expect(allWords.includes('BATON')).toBeFalsy()
+   })
```
As mentioned above, even with the regex working properly, some tests were failing.

This is because all of the strings are lower case in our list of words! Simply fixing the casing on your tests was enough to get them to pass after implementing the change I suggested to your wordOptions logic.

A-Tanner reviewed (changes requested):
Over all, great stuff! I noticed your hint button wasn't quite working, and I was able to fix it and explain my process. Be aware that the requested submission procedure is to use a separate branch (not your main) to write code and launch the pull request.

dcox2005 commented on wordle-web/components/word-options.vue:
```diff
@@ -0,0 +1,90 @@
+      Fill ? word<br>
+      (Cheating)
+    </v-btn>
+    <v-dialog v-model="dialog" width="500">
```
I too could not get the hint system to work.  When I click the button it always displays zero for me and when I click the zero, there are no words in the list to choose from.

dcox2005 commented on wordle-web/scripts/wordOptions.ts:
```diff
@@ -0,0 +1,37 @@
+     }
+
+     let expression: string = '^[A-Z]{0,5}'
+     for (let i = 0; i < 5; i++) {
```
your regex might also be a problem.  Regex is case sensitive, and this is looking for only capital letters, but if you look at the list of words we are checking against, the words are all in lower case letters. Unless somewhere you are using an ignore case and I'm not seeing it.

dcox2005 reviewed (commented):
Overall, I feel you have a nice start here.  There are some things still to work on.  Visually, when in full screen on the game page the keyboard still appears on the right of the game board.  We should make it like wordle and appear below it.  I like the sounds that play with the buttons.

AntonioGomez12 commented on wordle-web/components/keyboard.vue:
```diff
@@ -77,5 +82,16 @@ export default class KeyBoard extends Vue {
 
     return Letter.getColorCode(LetterStatus.Unknown)
   }
+  makeSound(){
+    let sound = new Audio("http://peal.io/download/feju3")
+
+    sound.currentTime=0;
+    sound.play()
+    setInterval(function(){
+    if(sound.currentTime> 1){
+      sound.pause();
+    }
+    },1);
+  }
```
Thanks for all the information! I'll be sure to fix it all today before the final submission!

AntonioGomez12 commented:
> link: https://orange-flower-0da9ed11e.1.azurestaticapps.net
> Extra credit: added sound to keyboard buttons, added help bar to clarify color meanings and how to play.
> partner: Antonio Gomez

mmwoodfo commented:
Make sure you create a separate branch for your assignment, rather than making it from your main branch.

mmwoodfo commented:
Be sure to run the command `npm run lintfix` to auto fix any formatting or linter errors.

If you ran your linter, you would also be able to quickly spot issues such as these:
![image](https://user-images.githubusercontent.com/43832670/167058962-f69c7813-a3f9-4293-9a36-cef950776225.png)

mmwoodfo commented on wordle-web/components/settings-dialog.vue:
```diff
@@ -1,9 +1,91 @@
+          <v-card-title> Menu </v-card-title>
+
+          <v-card-text>
+            <v-menu offset-y>
+              <template #activator="{ on, attrs }">
+                <v-btn color="primary" dark v-bind="attrs" v-on="on" nuxt to="/about">
+                  About
+                </v-btn>
+              </template>
+            </v-menu>
```
v-menu is not typically used in this manner. Take a look at this PR comment, as it's a similar problem!
https://github.com/IntelliTect-Samples/EWU-CSCD379-2022-Spring/pull/38/files#r865522913

mmwoodfo reviewed (commented):
This is super awesome. Good job on the extra credit as well!

mmwoodfo commented:
![image](https://user-images.githubusercontent.com/43832670/167060055-e7a03d9d-9c6f-4b4d-86c3-202fb0cc87b9.png)

So, no partner?
